src/auto/pad_controller.rs

```diff
@@ -0,0 +1,121 @@
+// This file was generated by gir (8080733) from gir-files (469db10)
+// DO NOT EDIT
+
+use EventController;
+#[cfg(any(feature = "v3_22", feature = "dox"))]
+use PadActionType;
+#[cfg(any(feature = "v3_22", feature = "dox"))]
+use Window;
+use ffi;
+use gdk;
+use gio;
+use glib;
+use glib::StaticType;
+use glib::Value;
+use glib::object::Downcast;
+use glib::object::IsA;
+use glib::signal::SignalHandlerId;
+use glib::signal::connect;
+use glib::translate::*;
+use glib_ffi;
+use gobject_ffi;
+use std::boxed::Box as Box_;
+use std::mem;
+use std::mem::transmute;
+use std::ptr;
+
+glib_wrapper! {
+    pub struct PadController(Object<ffi::GtkPadController, ffi::GtkPadControllerClass>): EventController;
+
+    match fn {
+        get_type => || ffi::gtk_pad_controller_get_type(),
+    }
+}
+
+impl PadController {
+    #[cfg(any(feature = "v3_22", feature = "dox"))]
+    pub fn new<'a, P: IsA<Window>, Q: IsA<gio::ActionGroup>, R: IsA<gdk::Device> + 'a, S: Into<Option<&'a R>>>(window: &P, group: &Q, pad: S) -> PadController {
+        skip_assert_initialized!();
+        let pad = pad.into();
+        let pad = pad.to_glib_none();
+        unsafe {
+            from_glib_full(ffi::gtk_pad_controller_new(window.to_glib_none().0, group.to_glib_none().0, pad.0))
+        }
+    }
+}
+
+pub trait PadControllerExt {
+    #[cfg(any(feature = "v3_22", feature = "dox"))]
+    fn set_action(&self, type_: PadActionType, index: i32, mode: i32, label: &str, action_name: &str);
+
+    //#[cfg(any(feature = "v3_22", feature = "dox"))]
+    //fn set_action_entries(&self, entries: /*Ignored*/&[&PadActionEntry]);
```

Comment:
Strange, file generated but it still ignored.
Maybe you forgot remove empty file?

Reply:
Yes, I can't generate `PadActionEntry` because memory management functions are missing.